srcts/src/components/tooltip.ts

```diff
@@ -147,35 +160,73 @@ export class BslibTooltip extends LightElement {
 
   private _toggle(x: ToggleMessage["value"]): void {
     if (x === "toggle") {
-      this._tooltip.toggle();
-    } else if (x === "show") {
-      if (!this.visible) this._tooltip.show();
-    } else if (x === "hide") {
-      if (this.visible) this._tooltip.hide();
+      x = this.visible ? "hide" : "show";
+    }
+    if (x === "hide") {
+      this._tooltip.hide();
     }
+    if (x === "show") {
+      this._show();
+    }
+  }
+
+  // No-op if the tooltip is already visible or if the trigger element is not visible
+  // (in either case the tooltip likely won't be positioned correctly anyway)
+  private _show(): void {
+    if (!this.visible && this.visibleTrigger) {
+      this._tooltip.show();
+    }
+  }
+
+  get visibleTrigger(): boolean {
+    const el = this.triggerElement as HTMLElement;
+    return el && el.offsetParent !== null;
```

Comment:
Make sure behavior is sensible when showing tooltip that isn't in the scroll area